### PR TITLE
fix(devSrv): Improve dev-server reliability

### PR DESCRIPTION
**Pull request by dr1rrb**
closes https://github.com/unoplatform/uno/issues/17753
closes https://github.com/unoplatform/uno/issues/17756
linked https://github.com/unoplatform/uno/issues/17757
closes https://github.com/unoplatform/uno/issues/17846

## Bugfix
Improve dev-server reliability

## What is the current behavior?
* Dev-Server is restarted only after build, driving the built app to not have the updated port to use
* When dev-server uses a new port, app needs to be manullay rebuilt
* Connection cannot be restore

## What is the new behavior?
* Dev-server is automatically restarted 5sec after it's being closed
* Dev-server attempts to re-use the same port when possible, allowing apps to restore connection automatically
* On re-connection, the client will make sure to re-configure the server (discover processors)

## PR Checklist
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

**Comment by dr1rrb** [comment]
@Mergifyio backport release/stable/5.3

File: src/Uno.UI.RemoteControl.VS/EntryPoint.cs [hunk 1]
```diff
@@ -374,6 +381,17 @@ private async Task StartServerAsync()
 		{
 			_processGate.Release();
 		}
+
+		async Task Restart()
```
Comment (jeromelaban) [inline src/Uno.UI.RemoteControl.VS/EntryPoint.cs:385]:
```suggestion
		private async Task Restart()
```

File: src/Uno.UI.RemoteControl.VS/EntryPoint.cs [hunk 2]
```diff
@@ -374,6 +381,17 @@ private async Task StartServerAsync()
...
+		{
+			if (_closing || _ct.IsCancellationRequested)
+			{
+				return;
+			}
+
+			await Task.Delay(5000, _ct.Token);
```
Comment (jeromelaban) [inline src/Uno.UI.RemoteControl.VS/EntryPoint.cs:392]: Let's log that we found the server stopped.

**Review by jeromelaban — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.3

**Pull request merged**